### PR TITLE
fix(sql): correctly escape double quotes when exporting CSVs

### PR DESCRIPTION
**Pull request by nwoolmer**
Fixes https://github.com/questdb/questdb/issues/1716
Fixes https://github.com/questdb/questdb/issues/3719

This fixes an old issue with CSV export, where JSON-style string escaping is used i.e escaping double quotes `"` using a blackslash `\`.

However, [the RFC](https://www.ietf.org/rfc/rfc4180.txt) suggests that strings containing `"` should be enclosed in an outer `"` pair, and then escaped with an extra `"` in front.

This has also been manually tested, so you can round trip queries like this:

SQL:

```sql
SELECT '"FOO"' AS FOO
```

Before:

```csv
"FOO"
"\"FOO\""
```

image

After:

```csv
"FOO"
"""FOO"""
```

image

### CODE REVIEW

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 34 (88.24%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/Utf8Sink.java|28|32|87.50%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|2|2|100.00%|

File: core/src/test/java/io/questdb/test/cutlass/http/IODispatcherTest.java [hunk 1]
```diff
@@ -7400,6 +7400,89 @@ public void testTextQueryCopyFrom() throws Exception {
...
+
+        new HttpQueryTestBuilder()
+                .withTempFolder(root)
+                .withMicrosecondClock(new TestMicroClock(0, 0))
+                .withWorkerCount(1)
+                .withHttpServerConfigBuilder(new HttpServerConfigurationBuilder())
+                .run((engine, sqlExecutionContext) -> {
+                    sendAndReceive(
+                            NetworkFacadeImpl.INSTANCE,
+                            "GET /exp?query=SELECT%20%27%7B%22filed1%22%3A1%2C%20%22filed2%22%3A1%2C%20%22filed3%22%3A%22admin%22%2C%20%22filed4%22%3A1%7D%27+as+foo HTTP/1.1\r\n" +
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/cutlass/http/IODispatcherTest.java:7414]: nit: we have `urlEncodeQuery()` utility method to keep the query human-readable

File: core/src/main/java/io/questdb/cutlass/http/processors/TextQueryProcessor.java [hunk 2]
```diff
@@ -304,9 +304,9 @@ private static void putUuidOrNull(HttpChunkedResponse response, long lo, long hi
         Numbers.appendUuid(lo, hi, response);
     }
 
-    private static void putVarcharOrNull(HttpChunkedResponse r, Utf8Sequence us) {
+    private static void putVarcharOrNull(HttpChunkedResponse response, Utf8Sequence us) {
         if (us != null) {
-            r.putQuote().escapeJsonStr(us).putQuote();
+            response.putQuote().escapeCsvStr(us).putQuote();
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/cutlass/http/processors/TextQueryProcessor.java:309]: We have one more place in this class where `escapeJsonStr()` is used:
```java
response.putQuote().escapeJsonStr(state.metadata.getColumnName(state.columnIndex)).putQuote();
```

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 31 / 35 (88.57%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/Utf8Sink.java|28|32|87.50%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|3|3|100.00%|

File: core/src/main/java/io/questdb/std/str/Utf8Sink.java [hunk 3]
```diff
@@ -30,6 +30,64 @@
 
 public interface Utf8Sink extends CharSink<Utf8Sink> {
 
+    /*
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/std/str/Utf8Sink.java:33]: Would be great to remove the old comment.

**Review by puzpuzpuz — DISMISSED** [review]

**Pull request merged**